runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -1390,6 +1390,10 @@ TR::TreeLowering::lowerStoreArrayElement(TR::PreorderNodeIterator& nodeIter, TR:
    arraylengthNode->setArrayStride(dataWidth);
 
    //ILGen for array element store already generates a NULLCHK
+   //If the helper call node is anchored under NULLCHK due to compactNullChecks, the NULLCHK is split into the helper call block.
+   //The NULLCHK is required for regular array block.
+   if (!anchoredArrayBaseAddressNode->isNonNull() && tt->getNode()->getOpCodeValue() == TR::NULLCHK)
+      arrayStoreCHKTT->insertBefore(TR::TreeTop::create(comp, TR::Node::createWithSymRef(TR::NULLCHK, 1, 1, arraylengthNode, comp->getSymRefTab()->findOrCreateNullCheckSymbolRef(comp->getMethodSymbol()))));
```

Comment:
@vijaysun-omr During tests, I found the `jitStoreFlattenableArrayElement` call node could be compacted under a `NULLCHK`. Any chance these call nodes (`jit{Load|Store}FlattenableArrayElement`) might be compacted into other types of opcodes other than `NULLCHK`?

Reply:
I added ASSERTs in a067353 to catch any case where the treetop node is neither a `treetop` nor `NULLCHK`.

Reply:
I doubt these trees can be compacted with other treetops that you asked about @a7ehuo since I don't believe there is any indirection expressed in the trees.